commands/delete-key.js

```diff
@@ -20,6 +22,19 @@ async function deleteAccessKey(options) {
     console.log(`Deleting key = ${options.accessKey} on ${options.accountId}.`);
     const near = await connect(options);
     const account = await near.account(options.accountId);
-    const result = await account.deleteKey(options.accessKey);
-    inspectResponse.prettyPrintResponse(result, options);
+    let accessKeys = await account.getAccessKeys();
+    let fullAccessKeys = accessKeys.filter(accessKey => accessKey.access_key.permission === 'FullAccess');
+
+    if (fullAccessKeys.length === 1 && fullAccessKeys[0].public_key.includes(options.accessKey)){
+        const answer = await askYesNoQuestion(
+            chalk`{bold.white You want to remove the last full access key and forgot access to this account ? {bold.green (y/n)}}`,
+            false
+        );
+        if (answer) {
+            const result = await account.deleteKey(options.accessKey);
+            inspectResponse.prettyPrintResponse(result, options);
+        } else {
+            console.log('Deleting key canceled.');
+        }
+    }
```

Comment:
Hmm, this means that the key will **only ever** get deleted if the account only has one full access key (even if the to-be-deleted key is of a completely different type). Let's follow @volovyk-s's [advice](https://github.com/near/near-cli/issues/834#issuecomment-968970448) here and make the flow look something like this:

```suggestion
    if (await checkIfSafeToDelete(account, options.accessKey)) {
        const result = await account.deleteKey(options.accessKey);
        inspectResponse.prettyPrintResponse(result, options);
    }
```

where `checkIfSafeToDelete` is a function that returns true if either the key is not a full access key or if the user agreed to delete it (twice if it is the last full access key). Something like this:
```js
async function checkIfSafeToDelete(account, accessKey) {
    let accessKeys = await account.getAccessKeys();
    let fullAccessKeys = accessKeys.filter(accessKey => accessKey.access_key.permission === 'FullAccess');

    if (fullAccessKeys.find(key => key.public_key === accessKey)) {
        const answer = await askYesNoQuestion(
            chalk`{bold.white Key [ {bold.blue ${accessKey}} ] is a full access key. Make sure it's not your recovery method. Do you want to proceed? {bold.green (y/n) }}`,
            false
        );
        if (!answer) {
            return false;
        }
        if (fullAccessKeys.length === 1) {
            const answer = await askYesNoQuestion(
                chalk`{bold.white Key [ {bold.blue ${accessKey}} ] is the last full access key. In case of deletion, you will not be able to restore access to the account [ {bold.blue ${accountId}} ]. Do you want to proceed? {bold.green (y/n) }}`,
                false
            );
            if (!answer) {
                return false;
            }
        }
    }
    return true;
}
```